package/resource_monitor/resource_monitor/src/query_process_file_descriptor.c

```diff
@@ -0,0 +1,491 @@
+    msg_linux_process_fd_count_t *fd_counts = (msg_linux_process_fd_count_t *)sbp_buf;
+    fd_counts->index = (u8)index;
+    fd_counts->pid = entry->pid;
+    fd_counts->fd_count = entry->fd_count;
+#ifdef DEBUG_QUERY_FD_TAB
+    PK_LOG_ANNO(LOG_DEBUG,
+                "index : %d pid: %d : fd_count : %d\n",
+                fd_counts->index,
+                fd_counts->pid,
+                fd_counts->fd_count);
+#endif
+    *len = sizeof(msg_linux_process_fd_count_t);
+  } break;
+  case SEND_FD_SUMMARY:
+    *msg_type = SBP_MSG_LINUX_PROCESS_FD_SUMMARY;
+    msg_linux_process_fd_summary_t *fd_summary = (msg_linux_process_fd_summary_t *)sbp_buf;
+    fd_summary->sys_fd_count = state->total_fd_count;
+    int nBytes = 0;
+    for (int i = 0; i < ITEM_COUNT; i++) {
+
+      unsigned int buf_space =
+        (unsigned int)(SBP_FRAMING_MAX_PAYLOAD_SIZE - sizeof(msg_linux_process_fd_summary_t)
+                       - (unsigned int)nBytes);
+      file_entry_t *file_entry = file_table_get(state->file_table, state->top10_file_counts[i]);
+      nBytes += snprintf(&fd_summary->most_opened[nBytes],
+                         buf_space,
+                         "%d%c%s",
+                         file_entry->fd_count,
+                         '\0',
+                         file_entry->file_str);
+#ifdef DEBUG_QUERY_FD_TAB
+      PK_LOG_ANNO(LOG_DEBUG,
+                  "Summary:%d : %d %s bytes",
+                  nBytes,
+                  file_entry->fd_count,
+                  file_entry->file_str);
+#endif
+      if ((SBP_FRAMING_MAX_PAYLOAD_SIZE - sizeof(msg_linux_process_fd_summary_t) - 1)
+          < (unsigned int)nBytes) {
+        nBytes = (SBP_FRAMING_MAX_PAYLOAD_SIZE - sizeof(msg_linux_process_fd_summary_t) - 1);
+        break;
+      }
+    }
+    fd_summary->most_opened[nBytes] = '\0';
+    *len = (u8)(sizeof(msg_linux_process_fd_summary_t) + (unsigned)nBytes + 1);
+    break;
+  case SEND_DONE:
+  default: {
+#ifdef DEBUG_QUERY_FD_TAB
+    PK_LOG_ANNO(LOG_ERR, "file descriptor query: Invalid state value: %d", state->send_state);
+#endif
+    return false;
+  }
+  }
+  state->send_state++;
+  return true;
+}
+
+static void *init_resource_query()
+{
+  u16 pid_count = 0;
+
+  resq_state_t *state = calloc(1, sizeof(resq_state_t));
+  resq_read_property_t read_prop = {.id = QUERY_SYS_PROP_PID_COUNT, .type = RESQ_PROP_U16};
+
+  if (!resq_read_property(QUERY_SYS_STATE_NAME, &read_prop)) {
+#ifdef DEBUG_QUERY_FD_TAB
+    PK_LOG_ANNO(LOG_ERR,
+                "failed to read 'pid count' property from '%s' module",
+                QUERY_SYS_STATE_NAME);
+#endif
+    goto error;
+  }
+
+  pid_count = read_prop.property.u16;
+#ifdef DEBUG_QUERY_FD_TAB
+  PK_LOG_ANNO(LOG_ERR,
+              "'%s' module reported %d processes on the system",
+              QUERY_SYS_STATE_NAME,
+              pid_count);
+#endif
+
+  if (pid_count > MAX_PROCESS_COUNT) {
+#ifdef DEBUG_QUERY_FD_TAB
+    PK_LOG_ANNO(LOG_ERR,
```

Comment:
Should leave this enabled, again, this is a degenerate state that should happen, if we the system is in this state we should squawk about it.  Let's change the priority to `LOG_ERR | LOG_SBP` so that the error goes into the SBP log as well.